lua/packer/compile.lua

```diff
@@ -41,21 +41,28 @@ end
 ]]
 
 local module_loader = [[
+
+local function escape_pattern(text)
+    return text:gsub("([^%w])", "%%%1")
+end
+
+local lazy_load_called = {}
+local to_load = {}
+
 local function lazy_load_module(module_name)
-  if module_name == 'packer.load' then return nil end
-  local to_load = {}
-  local i = 1
+  if module_name == 'packer.load' or lazy_load_called[module_name] then return nil end
+  lazy_load_called[module_name] = true
   for module_pat, plugin_name in pairs(module_lazy_loads) do
-    if not _G.packer_plugins[plugin_name].loaded and string.match(module_name, "^" .. module_pat) then
-      to_load[i] = plugin_name
+    if not _G.packer_plugins[plugin_name].loaded and string.match(module_name, "^" .. escape_pattern(module_pat)) then
+      to_load[#to_load + 1] = plugin_name
     end
   end
 
   require('packer.load')(to_load, {module = module_name}, _G.packer_plugins)
 end
 
 if not vim.g.packer_custom_loader_enabled then
-  table.insert(package.loaders, 2, lazy_load_module)
+  table.insert(package.loaders, 1, lazy_load_module)
```

Comment:
I want to try one thing to potentially let us put this loader second; I'll experiment and push a commit if I get it working.